public/js/prompt-methods.js

```diff
@@ -111,6 +168,32 @@ async function createButtonsForPrompts(
   }
 }
 
+/**
+ * Creates buttons for prompts dynamically based on available API interventions.
+ *
+ * This method generates buttons for a given type of prompt by filtering the 
+ * API response (`availableInterventions`) and creating buttons with human-readable 
+ * labels for each intervention. It is intended for prompts that are dynamically 
+ * fetched from the server, as opposed to hardcoded or static configurations.
+ *
+ * @param {HTMLElement} buttonContainer - The container element where buttons will be appended.
+ * @param {Array<Object>} availableInterventions - Array of available interventions returned by the API.
+ * @param {string} promptConfigurationType - The type of prompt (e.g., "TYPE_REWARDED_AD", "TYPE_NEWSLETTER").
+ * @returns {Promise<void>} Resolves when buttons are created and added to the DOM.
+ */
+async function createButtonsForAvailablePrompts(
+  buttonContainer, filteredInterventions
+) {
+  filteredInterventions.forEach((intervention, index) => {
```

Comment:
I think if your intention is to have this be an async function at the top level, that the callback used in the `forEach` loop would also have to be async, so that you could await below for the resolution of `createButtonForAvailablePrompt`.  I don't think it's impacting the code right now, as the `forEach` loop isn't collecting the results like a `.map()` would, so the code is executing and operating on the DOM as it wants to, but I would say that either `createButtonForAvailablePrompt` should be awaited, or `createButtonsForAvaialblePrompts` should not be async.

Reply:
I've refactored this a bit and changed it to a traditional for loop, but have awaited on the createButtonForAvailablePrompt, which I think should be all good now.